### PR TITLE
[act,mkt,pay,agents] adapt to Market API in ya-client

### PR DESCRIPTION
**Pull request by tworec**
Adapts to golemfactory/ya-client#52.

Confirmed to be back-compatible with `yapapi v0.3.0`

### CODE REVIEW

**Review by mfranciszkiewicz — COMMENTED** [review]
`requestor-sdk` moved to https://github.com/golemfactory/yarapi

(PR for removing it from the main repo https://github.com/golemfactory/yagna/pull/759)

**Review by Wiezzel — APPROVED** [review]
:+1:

**Review by mfranciszkiewicz — APPROVED** [review]

File: core/payment/src/api/provider.rs [hunk 1]
```diff
@@ -70,16 +70,7 @@ async fn issue_debit_note(
...
-        != agreement
-            .offer
-            .provider_id
-            .clone()
-            .unwrap()
-            .parse()
-            .unwrap()
-    {
-        // FIXME: provider_id shouldn't be an Option
+    if &node_id != agreement.provider_id() {
```
Comment (nieznanysprawiciel) [inline core/payment/src/api/provider.rs:73]: :smiley:

**Review by nieznanysprawiciel — APPROVED** [review]

File: agent/provider/src/market/provider_market.rs [hunk 2]
```diff
@@ -66,7 +66,7 @@ pub struct AgreementApproved {
 struct Subscription {
     id: String,
     preset: Preset,
-    offer: Offer,
+    offer: DemandOfferBase,
```
[the remark this answers is not in the document]
Comment (tworec) [reply agent/provider/src/market/provider_market.rs:69]: Will address in next PR

**Pull request merged**